app/app_test.go

```diff
@@ -200,10 +247,53 @@ func TestDeploymentModelCreateDeployment(t *testing.T) {
 			ds := NewDeployments(&db, fs, "")
 
 			mockInventoryClient := &inventory_mocks.Client{}
-			mockInventoryClient.On("Search", ctx,
-				"tenant_id",
-				mock.AnythingOfType("model.SearchParams"),
-			).Return(testCase.InvDevices, testCase.TotalCount, testCase.SearchError)
+			if testCase.InputConstructor != nil && testCase.InputConstructor.Name != nil && len(testCase.InputConstructor.Devices) == 0 {
+				mockInventoryClient.On("Search", ctx,
+					"tenant_id",
+					model.SearchParams{
+						Page:    1,
+						PerPage: PerPageInventoryDevices,
+						Filters: []model.FilterPredicate{
+							{
+								Scope:     InventoryGroupScope,
+								Attribute: InventoryGroupAttributeName,
+								Type:      "$eq",
+								Value:     *testCase.InputConstructor.Name,
+							},
+							{
+								Scope:     InventoryIdentityScope,
+								Attribute: InventoryStatusAttributeName,
+								Type:      "$eq",
+								Value:     InventoryStatusAccepted,
+							},
+						},
+					},
+				).Return(testCase.InvDevices, testCase.TotalCount, testCase.SearchError)
+
+				if testCase.TotalCount > len(testCase.InvDevices) {
+					mockInventoryClient.On("Search", ctx,
+						"tenant_id",
+						model.SearchParams{
+							Page:    2,
+							PerPage: PerPageInventoryDevices,
```

Comment:
I do not fully understand, what is the problem with more than one page in search results? also: why we simulate two pages with PerPage equal PerPageInventoryDevices and not 1?

Reply:
ok, made it a bit more explicit!

Reply:
PerPageInventoryDevices is constant to 512, I do not want to write 512 devices in the test, and the test for the "next" page is simply: I've got less devices than the total, so providing one device instead of two is enough to trigger next page.

Reply:
right! (I thought about setting PerPage=1). but it is fine I guess.